repo_test.go

```diff
@@ -543,9 +546,18 @@ func (rs *RepoSuite) TestSign(c *C) {
 		s := &data.Signed{}
 		c.Assert(json.Unmarshal(rootJSON, s), IsNil)
 		c.Assert(s.Signatures, HasLen, len(keyIDs))
-		for i, id := range keyIDs {
-			c.Assert(s.Signatures[i].KeyID, Equals, id)
+
+		// Signatures may be in any order, so must sort key IDs before comparison.
+		wantKeyIDs := append([]string{}, keyIDs...)
+		sort.Strings(wantKeyIDs)
+
+		gotKeyIDs := []string{}
+		for _, sig := range s.Signatures {
+			gotKeyIDs = append(gotKeyIDs, sig.KeyID)
 		}
+		sort.Strings(gotKeyIDs)
+
+		c.Assert(wantKeyIDs, DeepEquals, gotKeyIDs)
```

Comment:
client_test approaches this with https://github.com/theupdateframework/go-tuf/blob/f38e930df33905abc5bb7be5fcd0668231807a88/util/util.go#L261

Reply:
This isn't quite the same as comparing sorted lists, right? For example:
```
a := []string{"AAA", "BBB", "CCC"}
b := []string{"AAA, "BBB", "CCC", "CCC"}

aSet := StringSliceToSet(a) 
bSet := StringSliceToSet(b)

// This evaluates to true, but the slices we're comparing are not equal.
reflect.DeepEqual(aSet, bSet)
```

I think in this case, we would want tests to fail if signatures are erroneously duplicated. Comparing sorted lists accomplishes this, but comparing sets does not.

Reply:
good point with duplicates. i'll have to look into that if that test makes sense to me now too